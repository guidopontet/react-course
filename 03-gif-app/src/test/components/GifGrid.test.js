const { shallow } = require("enzyme");

const { default: GifGrid } = require("../../components/GifGrid");
const { useFetchGifs } = require("../../hooks/useFetchGifs");
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid', () => {
  const category = 'Dragon Ball';

  test('should render component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={ category }/>);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images are loaded', () => {
    const gifs = [{
      id: 'asd',
      url: 'https://example.com/image.png',
      title: 'Example',
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={ category }/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
