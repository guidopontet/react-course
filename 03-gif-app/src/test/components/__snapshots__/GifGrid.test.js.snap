// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should render component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid should show items when images are loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="asd"
      key="asd"
      title="Example"
      url="https://example.com/image.png"
    />
  </div>
</Fragment>
`;
